### PR TITLE
OCT-199 Add catalog disable event on error for GetProduct endpoint

### PR DESCRIPTION
**Pull request by RiffVentura**
The catalog's GetProduct endpoint throws an error if the catalog product selection is invalid
This PR adds similar error handling to what was done on other endpoints to prevent the endpoint from crashing

### CODE REVIEW

**Review by eoliveauakeneo — APPROVED** [review]

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetProductActionTest.php [hunk 1]
```diff
@@ -23,15 +24,18 @@ protected function setUp(): void
...
     }
 
     public function testItGetsProductByCatalogIdAndUuid(): void
     {
-        $catalogId = 'db1079b6-f397-4a6a-bae4-8658e64ad47c';
         $this->client = $this->getAuthenticatedPublicApiClient(['read_catalogs', 'read_products']);
-        $this->createCatalog($catalogId, 'Store US', 'shopifi');
-        $this->enableCatalog($catalogId);
+        $this->createCatalog(
+            id: 'db1079b6-f397-4a6a-bae4-8658e64ad47c',
```
Comment (hogren) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetProductActionTest.php:35]: For curiosity, why did you stop to use a variable for the catalog id ?

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetProductActionTest.php [hunk 2]
```diff
@@ -137,16 +143,68 @@ public function testItReturnsNotFoundWhenProductDoesNotExistForTheCatalog(): voi
...
+            name: 'Store US',
+            ownerUsername: 'shopifi',
+            isEnabled: false,
+        );
+
+        $product = $this->createProduct('blue', [new SetEnabled(true)]);
+        $productUuid = (string) $product->getUuid();
+
```
Comment (hogren) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetProductActionTest.php:156]: I think that is more sustainable with this little modification `$productUuid = $product->getUuid()->toString();`. What is your opinion ?

**Review by hogren — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Controller/Public/GetProductAction.php [hunk 3]
```diff
@@ -76,6 +77,9 @@ private function getCatalog(string $id, string $productUuid): Catalog
 
     /**
      * @return Product
+     * @throws CatalogDisabledException
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Controller/Public/GetProductAction.php:80]: This exception is not thrown in this function.

**Review by masacc — APPROVED** [review]

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetProductActionTest.php (hunk 1 above)
Comment (RiffVentura) [reply components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetProductActionTest.php:35]: Mainly for consistency with other tests (here or other files) and readability.
The variable adds too little value here

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetProductActionTest.php (hunk 2 above)
Comment (tseho) [reply components/catalogs/back/tests/Integration/Infrastructure/Controller/Public/GetProductActionTest.php]: Why do you think it's more sustainable ?

**Pull request merged**